services/wallet/router/router_v2.go

```diff
@@ -541,38 +542,57 @@ func (r *Router) getBalanceMapForTokenOnChains(ctx context.Context, input *Route
 
 	balanceMap = make(map[string]*big.Int)
 
+	chainError := func(chainId uint64, token string, intErr error) {
+		if err == nil {
+			err = fmt.Errorf("chain %d, token %s: %w", chainId, token, intErr)
+		} else {
+			err = fmt.Errorf("%s; chain %d, token %s: %w", err.Error(), chainId, token, intErr)
+		}
+	}
+
 	for _, chain := range selectedFromChains {
+		// check token existance
 		token := input.SendType.FindToken(r.tokenManager, r.collectiblesService, input.AddrFrom, chain, input.TokenID)
 		if token == nil {
+			chainError(chain.ChainID, token.Symbol, ErrTokenNotFound)
+			continue
+		}
+		// check native token existance
+		nativeToken := r.tokenManager.FindToken(chain, chain.NativeCurrencySymbol)
+		if nativeToken == nil {
+			chainError(chain.ChainID, chain.NativeCurrencySymbol, ErrNativeTokenNotFound)
 			continue
 		}
 
 		// add token balance for the chain
-		tokenBalance := big.NewInt(1)
-		if input.SendType == ERC1155Transfer {
+		var tokenBalance *big.Int
+		if input.SendType == ERC721Transfer {
+			tokenBalance = big.NewInt(1)
+		} else if input.SendType == ERC1155Transfer {
 			tokenBalance, err = r.getERC1155Balance(ctx, chain, token, input.AddrFrom)
 			if err != nil {
-				return nil, errors.CreateErrorResponseFromError(err)
+				chainError(chain.ChainID, token.Symbol, errors.CreateErrorResponseFromError(err))
 			}
-		} else if input.SendType != ERC721Transfer {
+		} else {
 			tokenBalance, err = r.getBalance(ctx, chain.ChainID, token, input.AddrFrom)
 			if err != nil {
-				return nil, errors.CreateErrorResponseFromError(err)
+				chainError(chain.ChainID, token.Symbol, errors.CreateErrorResponseFromError(err))
 			}
 		}
-		balanceMap[makeBalanceKey(chain.ChainID, token.Symbol)] = tokenBalance
-
-		// add native token balance for the chain
-		nativeToken := r.tokenManager.FindToken(chain, chain.NativeCurrencySymbol)
-		if nativeToken == nil {
-			return nil, ErrNativeTokenNotFound
+		// add only if balance is not nil
+		if tokenBalance != nil {
+			balanceMap[makeBalanceKey(chain.ChainID, token.Symbol)] = tokenBalance
 		}
 
+		// add native token balance for the chain
 		nativeBalance, err := r.getBalance(ctx, chain.ChainID, nativeToken, input.AddrFrom)
 		if err != nil {
-			return nil, errors.CreateErrorResponseFromError(err)
+			chainError(chain.ChainID, token.Symbol, errors.CreateErrorResponseFromError(err))
+		}
+		// add only if balance is not nil
+		if nativeBalance != nil {
```

Comment:
Quick question, is this new behavior of trying chains in succession covered by test(s)?

Reply:
Hi @ilmotta we don't have tests that simulate failed network calls, thus the answer is no, it's not covered.